zrml/swaps/src/tests.rs

```diff
@@ -1365,6 +1381,39 @@ fn swap_exact_amount_in_exchanges_correct_values_with_cpmm() {
     });
 }
 
+#[test]
+fn swap_exact_amount_in_fails_if_min_asset_amount_out_is_not_satisfied_with_cpmm() {
+    ExtBuilder::default().build().execute_with(|| {
+        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, true);
+        let expected_amount = 9900990100;
```

Comment:
Oh, regarding you suggestion of using the math functions: Yeah, that's an option too. Generally, I try to avoid this. There's a subtle difference between the two tests: In one case, you're testing that the function works with a particular, separately verified number. When using your suggestion, you're testing that the function works with whatever `calc_out_given_in` produces. The latter carries the risk that when we break `calc_out_given_in`, the test might not raise an error.

Reply:
"My" version, on the other hand, is super annoying, because we manually have to replace the concrete number when `calc_out_given_in` changes. :wink: However, I don't think we will ever change the price functions.